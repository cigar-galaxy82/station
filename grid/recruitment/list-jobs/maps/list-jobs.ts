/* eslint-disable jest/no-export, jest/valid-describe, jest/valid-title, jest/no-identical-title */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

const describeIf = (
  condition: boolean,
  ...args: Parameters<jest.Describe>
): void => (condition ? describe(...args) : describe.skip(...args));

export const listJobsTest = (
  provider: string,
  options?: RecordingProcessOptions
): void => {
  describe(`recruitment/list-jobs/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('ListJobs', () => {
      beforeAll(() => {
        superface = buildSuperfaceTest({
          profile: 'recruitment/list-jobs',
          provider,
          useCase: 'ListJobs',
        });
      });

      describe('when searching for published jobs', () => {
        it('performs correctly', async () => {
          const result = await superface.run(
            {
              input: {
                state: 'published',
              },
              testName: 'list published jobs',
            },
            options
          );

          expect(() => result.unwrap()).not.toThrow();
          expect(result).toMatchSnapshot();
        });
      });

      describe('when searching for draft jobs', () => {
        it('performs correctly', async () => {
          const result = await superface.run(
            {
              input: {
                state: 'draft',
              },
              testName: 'list draft jobs',
            },
            options
          );

          expect(() => result.unwrap()).not.toThrow();
          expect(result).toMatchSnapshot();
        });
      });

      describeIf(
        provider === 'breezy-hr',
        'when specified company does not exist',
        () => {
          let companyId: string;

          beforeAll(() => {
            companyId = process.env.BREEZY_HR_COMPANY_ID!;

            process.env.BREEZY_HR_COMPANY_ID = '1b111c1111ef11';
          });

          afterAll(() => {
            process.env.BREEZY_HR_COMPANY_ID = companyId;
          });

          it('returns error', async () => {
            const result = await superface.run(
              {
                input: {
                  state: 'published',
                },
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        }
      );

      describeIf(
        provider === 'workable',
        'when specified subdomain does not exist',
        () => {
          let subdomain: string;

          beforeAll(() => {
            subdomain = process.env.WORKABLE_SUBDOMAIN!;

            process.env.WORKABLE_SUBDOMAIN = 'invalid-superface';
          });

          afterAll(() => {
            process.env.WORKABLE_SUBDOMAIN = subdomain;
          });

          it('returns error', async () => {
            const result = await superface.run(
              {
                input: {
                  state: 'published',
                },
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        }
      );
    });
  });
};
