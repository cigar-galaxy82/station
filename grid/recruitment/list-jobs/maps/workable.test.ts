import { listJobsTest } from './list-jobs';

listJobsTest('workable');
