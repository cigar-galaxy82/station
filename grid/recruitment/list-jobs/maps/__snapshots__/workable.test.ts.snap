// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recruitment/list-jobs/workable ListJobs when searching for draft jobs performs correctly 1`] = `
Ok {
  "value": Object {
    "jobs": Array [
      Object {
        "benefits": "<p>Job Benefits</p>",
        "departmentName": null,
        "description": "<p>Job Description</p>",
        "id": "26053243E1",
        "name": "QA Tester",
        "requirements": "<p>Job Requirements</p>",
      },
    ],
    "pagination": Object {
      "nextPage": undefined,
    },
    "rateLimit": Object {
      "bucket": "workable",
      "remainingRequests": 8,
      "remainingRequestsPercentage": 80,
      "resetTimestamp": 1671628960,
      "totalRequests": 10,
    },
  },
}
`;

exports[`recruitment/list-jobs/workable ListJobs when searching for published jobs performs correctly 1`] = `
Ok {
  "value": Object {
    "jobs": Array [
      Object {
        "benefits": null,
        "departmentName": null,
        "description": "<p>We are looking for a Senior Software Engineer to produce and implement functional software solutions. You will work with upper management to define software requirements and take the lead on operational and technical projects. In this role, you should be able to work independently with little supervision. You should have excellent organization and problem-solving skills. If you also have hands-on experience in software development and agile methodologies, weâ€™d like to meet you. Your goal will be to develop high-quality software that is aligned with user needs and business goals.</p><p><br></p><h3>Responsibilities</h3><ul> <li>Develop high-quality software design and architecture</li> <li>Identify, prioritize and execute tasks in the software development life cycle</li> <li>Develop tools and applications by producing clean, efficient code</li> <li>Automate tasks through appropriate tools and scripting</li> <li>Review and debug code</li> <li>Perform validation and verification testing</li> <li>Collaborate with internal teams and vendors to fix and improve products</li> </ul>",
        "id": "FCFA80DF09",
        "name": "Senior Software Engineer",
        "requirements": "<ul> <li>Extensive experience in software development, scripting and project management</li> <li>Experience using system monitoring tools (e.g. New Relic) and automated testing frameworks</li> </ul>",
      },
      Object {
        "benefits": null,
        "departmentName": null,
        "description": "<p>We are looking for a high-performing Sales Director. The successful candidate will meet our customer acquisition and revenue growth objectives. Sales director responsibilities include developing key growth sales strategies, tactics and action plans. Successful execution of these strategies is required to achieve your financial targets. Sales director duties will include hitting annual targets, building relationships and understanding customer trends.</p><p><br></p><h3>Responsibilities</h3><ul> <li>  Own and hit/exceed annual sales targets within assigned territory and accounts   Develop and execute strategic plan to achieve sales targets and expand our customer base   Build and maintain strong, long-lasting customer relationships   Partner with customers to understand their business needs and objectives   Effectively communicate the value proposition through proposals and presentations   Understand category-specific landscapes and trends   Reporting on forces that shift tactical budgets and strategic direction of accounts              </li> <li>  Develop and execute strategic plan to achieve sales targets and expand our customer base   Build and maintain strong, long-lasting customer relationships   Partner with customers to understand their business needs and objectives   Effectively communicate the value proposition through proposals and presentations   Understand category-specific landscapes and trends   Reporting on forces that shift tactical budgets and strategic direction of accounts            </li> <li>  Build and maintain strong, long-lasting customer relationships   Partner with customers to understand their business needs and objectives   Effectively communicate the value proposition through proposals and presentations   Understand category-specific landscapes and trends   Reporting on forces that shift tactical budgets and strategic direction of accounts          </li> </ul>",
        "id": "AEA1D940CA",
        "name": "Sales Director",
        "requirements": "<ul> <li>Proven sales executive experience, meeting or exceeding targets</li> <li>Previous experience as a sales executive, sales manager or sales and marketing director</li> <li>Ability to communicate, present and influence all levels of the organization, including executive and C-level</li> <li>Proven ability to drive the sales process from plan to close</li> <li>Proven ability to articulate the distinct aspects of products and services</li> </ul>",
      },
    ],
    "pagination": Object {
      "nextPage": undefined,
    },
    "rateLimit": Object {
      "bucket": "workable",
      "remainingRequests": 8,
      "remainingRequestsPercentage": 80,
      "resetTimestamp": 1671628749,
      "totalRequests": 10,
    },
  },
}
`;

exports[`recruitment/list-jobs/workable ListJobs when specified subdomain does not exist returns error 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"Not Found\\",
  \\"detail\\": \\"Unable to get subdomain by name invalid-superface. Check that the subdomain with that name exists.\\",
  \\"code\\": \\"NotFound\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"workable\\"
  }
}
AST Path: definitions[0].statements[0].statements[0]
Original Map Location: Line 9, column 5",
}
`;
